frontend/subscriptions/index.js

```diff
@@ -0,0 +1,47 @@
+import { getCurrentRoute } from '@shopgate/pwa-common/helpers/router';
+import {
+  login, logout, disableLogin, userDataReceived$,
+} from '@shopgate/engage/user';
+import {
+  LoadingProvider, showModal, i18n,
+} from '@shopgate/engage/core';
+import {
+  requestSignInWithAppleAuthorization,
+  requestSignInWithAppleCredentialState,
+} from '../classes';
+import { signInWithApple$ } from '../streams';
+import { fetchAppleConfig } from '../actions';
+
+/**
+ * @param {Function} subscribe The subscribe function.
+ */
+export default (subscribe) => {
+  subscribe(userDataReceived$, async ({ dispatch, action }) => {
+    dispatch(fetchAppleConfig());
+    const result = await requestSignInWithAppleCredentialState(action.user.userIdentifier);
```

Comment:
This command is executed whenever user data comes in. There is no check if the user is logged in with "Sign in with Apple" or if the app supports the feature after all.

I don't think that this causes problems right now. In any situation where the command is executed while the user is in a not apple related login state, the underlying promise is rejected with different types of errors. But since this rejection is not handled, the error will come visible in the console.

So besides the rejection handling i would protect the call with some checks
- is "Sign in with Apple" supported by the app and merchant
- does the user data contain a "userIdentifier"

You should consider a new stream that emits after the user data was received, and the apple config was fetched from the pipeline. So you can guarantee that all valid information is present.

Reply:
Done.